source/std.tex

```diff
@@ -49,6 +49,15 @@
            ]{hyperref}
 \usepackage{memhfixc}     % fix interactions between hyperref and memoir
 
+
+%% Grammar extraction.
+\IfFileExists{./std-gram.ext}{
+\usepackage[extract-env={bnftab,simplebnf,bnf,bnfkeywordtab}]{extract}
+\def\PrintGrammar{\input{std-gram.ext}}}{
+\usepackage[active,header=false,handles=false,copydocumentclass=false,generate=std-gram.ext,extract-env={bnftab,simplebnf,bnf,bnfkeywordtab}]{extract}
+\def\PrintGrammar{Run again to include grammar summary}
```

Comment:
There might be a way to close the current file by hacking into the guts of the extract package to make it close the output file and stop extracting. Then we'll get the desired output on every pass. I can try that if you'd entertain that route.